packages/foam-vscode/src/features/panels/tags-explorer.ts

```diff
@@ -25,10 +25,10 @@ const feature: FoamFeature = {
     context.subscriptions.push(
       treeView,
       foam.tags.onDidUpdate(() => {
+        provider.refresh();
         treeView.title = baseTitle + ` (${foam.tags.tags.size})`;
       })
     );
-    foam.tags.onDidUpdate(() => provider.refresh());
```

Comment:
I actually didn't run into issues with tag count becoming out of sync during my testing, but I still made this change as I believe there could be a race condition. The title could be wrong if  `provider.refresh()` doesn't complete before title is generated. Thoughts?

Reply:
There is no race condition here because we don't use the provider to compute the number, but the `foam.tags` object directly (also this is the reason why you didn't run into issues in the first place)